esphome/components/online_image/online_image.cpp

```diff
@@ -0,0 +1,279 @@
+#include "online_image.h"
+
+#include "esphome/core/log.h"
+
+static const char *const TAG = "online_image";
+
+#include "image_decoder.h"
+
+#ifdef ONLINE_IMAGE_PNG_SUPPORT
+#include "png_image.h"
+#endif
+
+namespace esphome {
+namespace online_image {
+
+using display::Display;
+using image::ImageType;
+
+inline bool is_color_on(const Color &color) {
+  // This produces the most accurate monochrome conversion, but is slightly slower.
+  //  return (0.2125 * color.r + 0.7154 * color.g + 0.0721 * color.b) > 127;
+
+  // Approximation using fast integer computations; produces acceptable results
+  // Equivalent to 0.25 * R + 0.5 * G + 0.25 * B
+  return ((color.r >> 2) + (color.g >> 1) + (color.b >> 2)) & 0x80;
+}
+
+OnlineImage::OnlineImage(const std::string &url, int width, int height, ImageFormat format, ImageType type,
+                         uint32_t download_buffer_size)
+    : Image(nullptr, 0, 0, type),
+      buffer_(nullptr),
+      download_buffer_(download_buffer_size),
+      format_(format),
+      fixed_width_(width),
+      fixed_height_(height) {
+  this->set_url(url);
+}
+
+void OnlineImage::release() {
+  if (this->buffer_) {
+    ESP_LOGD(TAG, "Deallocating old buffer...");
+    this->allocator_.deallocate(buffer_, get_buffer_size_());
+    this->data_start_ = nullptr;
+    this->buffer_ = nullptr;
+    this->width_ = 0;
+    this->height_ = 0;
+    this->buffer_width_ = 0;
+    this->buffer_height_ = 0;
+    end_connection_();
+  }
+}
+
+bool OnlineImage::resize_(int width_in, int height_in) {
+  int width = this->fixed_width_;
+  int height = this->fixed_height_;
+  if (auto_resize_()) {
```

Comment:
> All uses of class members and member functions should be prefixed with `this->` to distinguish them from global functions in code review.
https://esphome.io/guides/contributing#codebase-standards

```suggestion
  if (this->auto_resize_()) {
```

Reply:
I thought for protected fields/methods (which have the "_" suffix) it was clear enough and not necessary, but I must have dreamed that exception.